### ISSUE TITLE
Contributor Build Walkthrough

### ISSUE THREAD

**Issue by SekretOne** [Rebirth-of-the-Night/Rebirth-Of-The-Night#752]
Create a guided doc to cover:

### Getting Started

- [ ] Git Checkout
- [ ] How to build
- [ ] How to run
  - [ ] Minecraft Launcher
  - [ ] MMC 
    - [ ] step by step on sym link 
  - [ ] ~~CurseForge~~ I don't think this a good fit for fast development

### Contribution
- [ ] What and where to push
- [ ] Recommended commit strategies

**Comment by SekretOne** [comment]
@kil0byt3 if you have any thoughts or insights

**Comment by kil0byt3** [comment]
symlink idea is great

**Comment by democat3457** [comment]
also imma link #368

**Comment by SekretOne** [comment]
> also imma link #368
> 
> and what's symlink for? is it for mmc exclusive people

MMC doesn't let you redirect the directory of where to look for your instance's content. You can use symbolic links to address this- making the MMC .minecraft folder actually 'redirect' to your local repo's `build/<type>` directory

**Comment by SekretOne** [comment]
oh and yes, this structure rebuild resolves the underlying need of #368  because you're no longer tracking the directory that minecraft runs- avoiding inadvertent write-backs that alter files.

**Comment by democat3457** [comment]
wait but if the game auto-generates a config such as special_ai.cfg (i.e., updating that mod, which adds a new config option) then the long list will be committed as well, since most git clients dont have the ability to stage hunks (and I don't see how you can tell if a new config option was added, since git doesn't exist in the build folder)

**Comment by SekretOne** [comment]
> wait but if the game auto-generates a config such as special_ai.cfg

Yes. If you are relying on first time run to generate the config file and then tweak it, then yes you need to run it *somewhere* and then explicitly add that to `src`.

You do now have to deliberately add things- but it prevents the accidental adding of things.
